cmd/artifact/command/push.go

```diff
@@ -15,21 +15,22 @@ import (
 
 func pushCommand(options *Options) *cobra.Command {
 	releaseManagerClient := httpinternal.Client{}
-
+	var idpURL, clientID, clientSecret string
 	command := &cobra.Command{
 		Use:   "push",
 		Short: "push artifact to artifact repository",
 		RunE: func(c *cobra.Command, args []string) error {
 			var artifactID string
 			var err error
 			ctx := context.Background()
+			authenticator := httpinternal.NewClientAuthenticator(clientID, clientSecret, idpURL)
+			releaseManagerClient.Auth = &authenticator
 
-			if releaseManagerClient.Metadata.AuthToken != "" {
```

Comment:
Notice the change in behaviour here. We cannot see a reason to start up the artifact call without credentials. Therefore we do not support it anymore. If this is wrong, please let us know.